include/net/ieee802154_radio.h

```diff
@@ -24,6 +24,14 @@ extern "C" {
  * @{
  */
 
+enum ieee802154_hw_caps {
+	IEEE802154_HW_FCS		= BIT(0),
+	IEEE802154_HW_PROMISC		= BIT(1),
+	IEEE802154_HW_FILTER		= BIT(2),
+	IEEE802154_HW_CSMA		= BIT(3),
```

Comment:
Do you also want to insert IEEE802154_HW_ACK_REQ for the device with the capability to perform TX->RX(ACK) sequences?

Reply:
Yes, this will be necessary, as this PR https://github.com/zephyrproject-rtos/zephyr/pull/4115 reminded me as well.